tools/write_pyi.py

```diff
@@ -218,7 +218,8 @@ def _formatvalue(value):
 
     fn_doc = base_method.__doc__ if base_method else fn.__doc__
     has_docs = gen_docs and fn_doc is not None
-    docs = 'r"""' + f"{fn_doc}" + '"""' if has_docs else ""
+    string_prefix = "r" if chr(92) in fn_doc else ""
```

Comment:
this is probably a better regexp `r'\\[^\\]'` should match single unescaped `\` only

Reply:
I'm not sure about that - it could miss some cases [like this one](https://github.com/sqlalchemy/alembic/blob/1ae44556907f14f92874f05d05242cb57bb0f855/alembic/operations/ops.py#L2346)?

Reply:
indeed that does not seem to work on that string.